extension/src/features/approve_pasted_psbt.rs

```diff
@@ -0,0 +1,136 @@
+use crate::{
+    components::{
+        select::{Select, SelectItem},
+        textarea::TextArea,
+    },
+    context::UserContext,
+    utils::{helpers::get_clipboard, storage::LocalStorage},
+};
+use anyhow::anyhow;
+use anyhow::Result;
+use signer::{
+    psbt_details::PSBTDetails,
+    signer::decode_psbt_and_sign,
+    storage::{SettingsStorage, UserStorage},
+};
+use std::str::FromStr;
+use yew::prelude::*;
+use yew_router::prelude::use_navigator;
+
+#[function_component(ApprovePastedPSBT)]
+pub fn approve_pasted_psbt() -> Html {
+    let password = use_context::<UserContext>()
+        .unwrap()
+        .password
+        .clone()
+        .unwrap_or_default();
+    let navigator = use_navigator().unwrap();
+    let storage = UserStorage::read(LocalStorage::default());
```

Comment:
You use `LocalStorage::default()` lots of times, what about creating it once at the beginning and reusing it across? 

`let local_storage = LocalStorage::default();`

and then you clone it when you need it if you need to pass the ownership

Reply:
Moved to RefCell 💯